src/main/java/Duke.java

```diff
@@ -1,10 +1,213 @@
+import duke.exception.DukeDescriptionException;
+import duke.exception.InvalidCommandException;
+import duke.tasks.Deadline;
+import duke.tasks.Event;
+import duke.tasks.Task;
+import duke.tasks.Todo;
+
+import java.time.LocalDate;
+import java.util.Scanner;
+import java.io.File;
+
+/**
+ * Represents the main working class of DukeProject.
+ */
 public class Duke {
+    /** Primary data structure to store the tasks. */
+    private TaskList taskList;
+    /** Storage to read/write task list from/into files. */
+    private Storage storage;
+    /** Data structure to store tasks aligned by dates.*/
+    private Calender calender;
+    /** Object that handles user interface and communicating with user. */
+    private Ui ui;
+    /** Primary I/O object used. */
+    private Scanner sc;
+
+    /**
+     * Execute commands to manipulate task list and calender. The method will create/delete task objects as required
+     * and store/remove them from all data structures as needed.
+     * @param command Command to be executed.
+     * @throws DukeDescriptionException If command requires a description and it not given.
+     * @throws BadDescriptionException If description provided does not match the format required by command.
+     * @throws BadDateException If date provided does not match format required by command.
+     */
+    public void readCommand(Command command) throws DukeDescriptionException,
+            BadDescriptionException, BadDateException {
+        switch (command) {
+        case LIST:
+            System.out.println("Here are all your tasks:");
+            taskList.printTask();
+            break;
+        case DONE:
+            try {
+                String strDone = sc.nextLine();
+                if (strDone.isEmpty()) {
+                    throw new DukeDescriptionException("Empty Description");
+                }
+                int done = Integer.parseInt(strDone.substring(1)); //there must be a space between command and input
+                if (done <= 0 || done > taskList.getSize()) {
+                    throw new BadDescriptionException("Description for done cannot be "
+                            + done);
+                }
+                taskList.getTask(done - 1).setDone(true);
+                System.out.println("Nice! I've marked this task as done: \n" +
+                        "  " + taskList.getTask(done - 1));
+                storage.writeListIntoFile(taskList.getList());
+            } catch (NumberFormatException e) {
+                throw new BadDescriptionException("Non-Integer");
+            }
+            break;
+        case TODO:
+            String todo = sc.nextLine();
+            if (todo.isEmpty()) {
+                throw new DukeDescriptionException("Empty Description");
+            }
+            Task taskToDo = new Todo(todo.substring(1));
+            taskList.addTask(taskToDo);
+            System.out.println("I've added this task: \n" +
```

Comment:
Hi, please note that the + sign should be on the next line with the String to add.